extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -72,16 +84,28 @@ class AmpYoutube extends AMP.BaseElement {
     iframe.width = this.width_;
     iframe.height = this.height_;
     this.element.appendChild(iframe);
-    /** @private {?Element} */
-    this.iframe_ = iframe;
 
-    // TODO(mkhatib, #2050): Use PlayerReady message for layout promise.
-    return loadPromise(iframe);
+    /** @private {!Element} */
+    this.iframe_ = iframe;
+    this.iframe_.id = this.frameId_;
+    const ytPlayerReadyPromise = this.getPlayerReadyPromise_();
+
+    return loadPromise(iframe).then(() => {
+      // Make sure the YT player is ready for this. For some reason YT player
+      // would send couple of messages but then stop. Waiting for a bit before
+      // sending the 'listening' event seems to fix that and allow YT
+      // Player to send messages continuously.
+      return timer.promise(300).then(this.listenToFrame_.bind(this));
+    }).then(ytPlayerReadyPromise);
```

Comment:
I could swear I've read it somewhere but couldn't find it. I've swapped it to wrap it in a function. @JustinPinner any idea if this pattern is allowed?